### DISCUSSION TITLE
To do list

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#424] [General]
Source|Issue                        | State|Comments
------|-----------------------------|------|-------------------
#399  |[Airmont][SMBIOS] Wrong offset|Unsolved / Closed|Past `manufacturer_id` field
#389  |[Intel][Gemini Lake] IMC, MCHBAR  |Unsolved|Registers are empty
#378  |[AMD][Raphael] SVI voltage   |Closed|Workaround released
#378  |[AMD][Raphael] HSMP unreachable|Opened|Protocol timeout
#378  |[AMD][Raphael] Kernel symbols|Closed|kernel updated
#374  |[Intel][Skylake/X] Non regression|Closed|Need processor
#366  |[Intel][Airmont] Z8000 IMC   |Closed|#399
#313  |[Intel][Cascade Lake/X] IMC  |Canceled|Need to develop on processor
#410  |[Intel] Hardware Duty Cycling|Unsolved / Closed| No HDC cycles on TGL
#418 |[zen-kernel](https://github.com/zen-kernel/zen-kernel) incompatible|Closed|Possible SMU conflict

**Comment by antermin** [comment]
Regarding #366, I have Celeron(R) CPU N3150, architecture shows Atom/Airmont.

Is there anything that I can help?

**Comment by cyring** [comment]
> Regarding #366, I have Celeron(R) CPU N3150, architecture shows Atom/Airmont.
> 
> Is there anything that I can help?

I believe we should rather read ["Braswell"](https://ark.intel.com/content/www/us/en/ark/products/codename/66094/products-formerly-braswell.html) ?

**EDIT**: Fixed and available in the [`develop`](https://github.com/cyring/CoreFreq/tree/develop) branch for your testings.

Can you create a new issue `Celeron(R) CPU N3150` then post details from _CoreFreq_:
```
corefreq-cli -s -n -m -n -B -n -M -n -C 1
```

**Comment by cyring** [comment]
[Spreadtrum SC9853I](https://github.com/pytorch/cpuinfo/blob/de2fa78ebb431db98489e78603e4f77c1f6c5c57/test/cpuinfo/leagoo-t5c.log) CPUID `06_75` perhaps Airmont [compatible](https://phonedb.net/index.php?m=processor&id=755&c=spreadtrum_sc9853i) ?

Edit: done.

**Reply by cyring** [reply]
[cpubenchmark](https://www.cpubenchmark.net/cpu.php?cpu=Spreadtrum+SC9853I-IA&id=4179)

[specdevice](https://specdevice.com/showspec.php?id=1de3-7f73-66cb-70280033c587)

[xda-developers](https://forum.xda-developers.com/t/joying-android-8-1-intel-airmont-eight-core-1-8ghz-sc9853i-4gb-ram-32gb-flash.3897206/)

**Comment by cyring** [comment]
# 13th Gen MCHBAR Memory Controller
## (part 3) (part 4) (part 5)

![image](https://user-images.githubusercontent.com/11563789/223085518-1766a249-7254-4d12-89be-368eab80bbe1.png)

* impacts RPL, probably ADL, decoder:
https://github.com/cyring/CoreFreq/blob/34efe5d3e6c79e5fddf78350da6fe4b7de0a63a8/corefreqk.c#L6145
* sub-channels are defined as a function of DDR Type

**Comment by cyring** [comment]
[EPYC Embedded 9004](https://www.tomshardware.fr/amd-epyc-embedded-9004-jusqua-96-coeurs-zen-4-pour-les-systemes-embarques/)

**Reply by cyring** [reply]
* Source: `Revision Guide for
 AMD Family 19h
 Models 10h-1Fh
 Processors`

CPUID Fn0000_0001_EAX|AMD EPYC™ 9004 Series Processors
-|-
`00A10F11h` `(Genoa-B1)`|X

**Reply by cyring** [reply]
Bergamo [screenshots](https://www.hardwaretimes.com/128-core-amd-epyc-bergamo-process-spotted-3-1ghz-boost-clock-and-a-tdp-of-360w/#:~:text=Bergamo%20is%20just%20like%20Genoa,128%2Dcore%20Epyc%209754%20flagship.) and [more](https://www.cnbeta.com.tw/articles/tech/1350149.htm)

**Comment by cyring** [comment]
As mentioned in ["_epyc-embedded-9004-series-product-brief_"](https://www.amd.com/system/files/documents/epyc-embedded-9004-series-product-brief.pdf) FOOTNOTES, I'm looking for "_Socket SP5/SP6 Platform RAS Specification and Implementation Guide for Family 19h Model 10-1Fh_" from `devhub.amd.com/wp-content/uploads/Docs/57065_0.95.pdf`

Anyone ?

**Comment by cyring** [comment]
EPYC Genoa #479 ; Ryzen 5 8600G #84

Refering to those [issues](https://github.com/cyring/CoreFreq/issues/479#issuecomment-2004606350); the remaining actions are:
* Decode geometry of new 24 and 48 GB DIMM. 
* Find registers for unknown power values
* Solve TjMax and offset values [_partial_]
* Debug detection of IOMMU [_done_]
* Improve 6 cores cluster mapping. #512

**Comment by cyring** [comment]
[Activity Monitors Unit (AMU)](https://www.kernel.org/doc/html/latest/arch/arm64/amu.html)
* CPU cycle counter: increments at the **frequency of the CPU**.
* Constant counter: increments at the fixed frequency of the system clock.
* Instructions retired
* Memory stall cycles

When in WFI or WFE these counters do not increment.

Prerequisite: ARMv**8.4**

**Comment by cyring** [comment]
# AArch64
* [`ID_AA64MMFR4_EL1`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch64-Registers/ID-AA64MMFR4-EL1--AArch64-Memory-Model-Feature-Register-4?lang=en)
* [`ID_AA64PFR2_EL1`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch64-Registers/ID-AA64PFR2-EL1--AArch64-Processor-Feature-Register-2?lang=en)
* [`ID_AA64FPFR0_EL1`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch64-Registers/ID-AA64FPFR0-EL1--AArch64-Floating-point-Feature-Register-0?lang=en)
* [`MVFR0_EL1 MVFR1_EL1 MVFR2_EL1`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch32-Registers/MVFR0--Media-and-VFP-Feature-Register-0?lang=en) `!32-bits`
* [`GMID_EL1`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch64-Registers/GMID-EL1--Multiple-tag-transfer-ID-Register?lang=en)
* [`DCZID_EL0`](https://developer.arm.com/documentation/ddi0601/2024-06/AArch64-Registers/DCZID-EL0--Data-Cache-Zero-ID-Register?lang=en)
* [`CPACR_EL1`](https://developer.arm.com/documentation/ddi0601/2024-09/AArch64-Registers/CPACR-EL1--Architectural-Feature-Access-Control-Register?lang=en)
* [`ID_AA64ISAR4_EL1`](https://gist.github.com/justtryingthingsout/2f22b9a5151c06c367bdd9b52b08b719#file-everest-core-sysregs-L201)

**Comment by cyring** [comment]
# 2025
## AMD Krackan Point
* AMD Ryzen AI 7 350
* AMD Ryzen AI 5 340
### CPUID
* [`BF_60h`](https://browser.geekbench.com/ai/v1/108829)

## AMD Fire Range, Strix Halo
* AI MAX+ 395
### CPUID
* [`BF_70h`](https://browser.geekbench.com/v6/cpu/9460276)

## AMD Granite Ridge
* AMD Ryzen 5 9600
## Intel Raptor Lake Refresh
* Intel Core 7 240H

## [Ryzen Z2](https://www.accessoires-asus.com/actualites/amd-z2-une-nouvelle-generation-de-processeurs-pour-le-gaming-mobile-1492.htm)
### Z2 Extreme
* 8 cœurs CPU (3 Zen 5 + 5 Zen 5c)
* GPU RDNA 3.5 avec 16 unités de calcul
* TDP de 35 Watts
* Performance GPU améliorée de 20% par rapport au Z1 Extreme
 
### Z2
* 8 cœurs CPU Zen 4
* GPU RDNA 3 avec 12 unités de calcul
* TDP de 30 Watts
* Similaire au Z1 Extreme en termes de performances
 
### Z2 Go
* 4 cœurs CPU Zen 3+
* GPU RDNA 2 avec 12 unités de calcul
* TDP de 30 Watts
* Conçu pour les consoles portables d'entrée de gamme

**Comment by cyring** [comment]
Dev ceased

**Reply by antermin** [reply]
Hello, does this mean that CoreFreq will no longer be developed (no more updates, bug fixes, etc.)?

**Reply by cyring** [reply]
Hello,
I would like the work to be continued by other developers.
I do not exclude the possibility of getting back into it but to a lesser extent and following the terms of its license.
Kind regards
